### PR TITLE
Add InstitutionID to topology Institutions

### PR DESCRIPTION
**Pull request by CannonLock**
? Means I need input
~ Means it wasn't available but also not needed
TODO Means it will be filled in pending OSG Institution creation

- [x] Add script to test that new ones have the field added
- [x] Wait for Janet to add the ones labeled TODO
- [x] Wait for Janet to inform on the ones with a ?

### CODE REVIEW

**Comment by CannonLock** [comment]
@matyasselmeci What is the deal with CancerComputer? Hasn't been touched for 6 years, can we delete?

**Comment by brianhlin** [comment]
@CannonLock 

> ? Means I need input

Could you provide the list in ticket here? It's hard to dig this up in the 220-file change

> What is the deal with CancerComputer? Hasn't been touched for 6 years, can we delete? Or are we confident enough in its death that I can mark its InstitutionID null for now?

Let's just use null for now. I'm not sure if there's been any contributions lately but I don't think we'd want to fully remove it

**Comment by CannonLock** [comment]
@brianhlin If you want a easy way to compare these entries with their names in the institution DB. 

They are null if they have no resources as I saw no reason to map them.

[facility_institution_mapping.txt](https://github.com/user-attachments/files/16910090/facility_institution_mapping.txt)

**Comment by osg-cat** [comment]
> @CannonLock
> 
> > ? Means I need input
> 
> Could you provide the list in ticket here? It's hard to dig this up in the 220-file change
> 
> > What is the deal with CancerComputer? Hasn't been touched for 6 years, can we delete? Or are we confident enough in its death that I can mark its InstitutionID null for now?
> 
> Let's just use null for now. I'm not sure if there's been any contributions lately but I don't think we'd want to fully remove it

Cancer Computer rebranded as “Computing for Humanity” a while ago: <https://www.computingforhumanity.com>.

File: .github/scripts/check_facility_has_institution_id/main.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+topology_institution_ids = {x['id'] for x in topology_institutions}
+
+def check_facility_institution_id(yaml_string: str):
+
+    facility = yaml.load(yaml_string, Loader=yaml.Loader)
+    if 'InstitutionID' not in facility:
+        raise Exception("FACILITY.yaml does not have an InstitutionID field")
+
+    if facility['InstitutionID'] not in topology_institution_ids and facility['InstitutionID'] is not None:
+        raise Exception(f"Invalid InstitutionID: {facility['InstitutionID']}")
```
Comment (matyasselmeci) [inline .github/scripts/check_facility_has_institution_id/main.py:23]: Looks like this script bails out on the first error. Can you make it print all errors instead and return nonzero if there were any?

File: topology/University of Wisconsin/WIPAC/IceCube.yaml [hunk 2]
```diff
@@ -7,8 +7,8 @@ Resources:
     Description: StashCache origin server for IceCube
     FQDN: stash-origin.icecube.wisc.edu
     ID: 1106
-    Services: 
-      XRootD origin server:
+    Services:
+      Pelican origin:
```
Comment (matyasselmeci) [inline topology/University of Wisconsin/WIPAC/IceCube.yaml:11]: This should probably be in a separate PR

File: topology/University of Wisconsin/WIPAC/IceCube.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ Resources:
     Description: StashCache origin server for IceCube
     FQDN: stash-origin.icecube.wisc.edu
     ID: 1106
-    Services: 
+    Services:
```
Comment (CannonLock) [inline topology/University of Wisconsin/WIPAC/IceCube.yaml]:
```suggestion
    Services: 
```

**Review by matyasselmeci — APPROVED** [review]
LGTM.
